sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -1002,179 +1007,115 @@ struct _pi_program : _pi_object {
     // is loaded via clCreateProgramWithBinary().
     Native,
 
-    // The program consists of native code (typically compiled from SPIR-v),
-    // but it has unresolved external dependencies which need to be resolved
-    // by linking with other Object state program(s).  Programs in this state
-    // have a single Level Zero module.
+    // The program was notionally compiled from SPIR-V form.  However, since we
+    // postpone compilation until the module is linked, the internal state
+    // still represents the module as SPIR-V.
     Object,
 
-    // The program consists of native code with no external dependencies.
-    // Programs in this state have a single Level Zero module, but no linking
-    // is needed in order to run kernels.
+    // The program has been built or linked, and it is represented as a Level
+    // Zero module.
     Exe,
 
-    // The program consists of several Level Zero modules, each of which
-    // contains native code.  Some modules may import external symbols and
-    // other modules may export definitions of those external symbols.  All of
-    // the modules have been linked together, so the imported references are
-    // resolved by the exported definitions.
-    //
-    // Module linking in Level Zero is quite different from program linking in
-    // OpenCL.  OpenCL statically links several program objects together to
-    // form a new program that contains the linked result.  Level Zero is more
-    // similar to shared libraries.  When several Level Zero modules are linked
-    // together, each module is modified "in place" such that external
-    // references from one are linked to external definitions in another.
-    // Linking in Level Zero does not produce a new Level Zero module that
-    // represents the linked result, therefore a program in LinkedExe state
-    // holds a list of all the pi_programs that were linked together.  Queries
-    // about the linked program need to query all the pi_programs in this list.
-    LinkedExe
```

Comment:
Should we retain this capability (dynamic linking) in the plugin in case any future uses of it arise?